onnxruntime/core/util/thread_utils.cc

```diff
@@ -98,7 +99,16 @@ CreateThreadPoolHelper(Env* env, OrtThreadPoolParams options) {
     }
     options.thread_pool_size = static_cast<int>(default_affinities.size());
     if (options.auto_set_affinity) {
+#ifdef _WIN32
```

Comment:
Even on servers setting affinity could result worse performance because normally each Windows server has a lot of background processes. Their activities will hurt the tail latency very much.

Reply:
Thread priorities still come into play whether on server or client.

I feel like the change as-is is safe & minimal. We also would like to get this change in ASAP upstream, for use on Windows client, and keeping it minimal and low impact helps there. 

This arose out of a conversation we had with @pranavsharma and others a few months ago regarding affinity use in ORT. At the time, we had decided it was ok for our team to own the default settings for ORT affinity on CLIENT (hence this PR). We also have more evidence now (via ETW OS traces) that indeed affinity was hurting power on big/little CPU designs.

While generally in Windows [we don't recommend setting affinity](https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-setthreadaffinitymask#remarks) on client or server, it's clear that ORT was previously more popular on server when this was first designed, and was likely thoroughly benchmarked and tested there. For now, I would rather not change the default on server where being on battery or having mixed big/little cores is not present. 

If there was more extensive testing on server, coordination with customers to ensure this doesn't regress server perf runtimes, and perf was the same or improved in all scenarios that we would certainly be supportive of the change.